x-pack/plugins/maps/public/connected_components/toolbar_overlay/fit_to_data/fit_to_data.tsx

```diff
@@ -15,24 +15,59 @@ interface Props {
   fitToBounds: () => void;
 }
 
-export const FitToData: React.FunctionComponent<Props> = ({ layerList, fitToBounds }: Props) => {
-  if (layerList.length === 0) {
-    return null;
+interface State {
+  canFit: boolean;
+}
+
+export class FitToData extends React.Component<Props, State> {
+  _isMounted: boolean = false;
+
+  state = { canFit: false };
+
+  componentDidMount(): void {
+    this._isMounted = true;
+    this._loadCanFit();
+  }
+
+  componentWillUnmount(): void {
+    this._isMounted = false;
+  }
+
+  componentDidUpdate(): void {
+    this._loadCanFit();
+  }
+
+  async _loadCanFit() {
+    const fittableLayersPromise = this.props.layerList.map((layer) => layer.isFittable());
```

Comment:
The naming is a little harder to follow in this method then it should be. How about naming `fittableLayersPromise` as just `promises` and then naming `isFittables` to just `canFit` and instead of doing a filter, use reduce and just `&&` each value together.